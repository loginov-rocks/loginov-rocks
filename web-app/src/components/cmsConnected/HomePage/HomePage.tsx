import * as React from 'react';

import { CmsComponent } from 'cms/interfaces/CmsComponent';
import { CmsConnectedProps } from 'cms/interfaces/CmsConnectedProps';

interface Props extends CmsConnectedProps {
  sections: CmsComponent[];
  title: string;
}

export const HomePage: React.FC<Props> = ({ render, sections, title }) => (
  <>
    <h1>{title}</h1>
    {render(sections)}
  </>
);
