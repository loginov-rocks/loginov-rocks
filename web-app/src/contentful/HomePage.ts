import { OpenSourceSection } from './OpenSourceSection';
import { RichTextSection } from './RichTextSection';
import { SocialPresenceSection } from './SocialPresenceSection';

type HomePageSection = OpenSourceSection | RichTextSection | SocialPresenceSection;

export interface HomePage {
  fields: {
    title: string;
    sections: HomePageSection[];
  };
  sys: {
    contentType: {
      sys: {
        id: 'homePage';
      };
    };
  };
}
