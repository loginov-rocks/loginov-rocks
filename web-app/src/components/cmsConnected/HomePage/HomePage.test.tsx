/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { HomePage } from './HomePage';

it('matches snapshot', () => {
  const wrapper = shallow(<HomePage render={() => null} sections={[]} title="Hello!" />);

  expect(wrapper).toMatchSnapshot();
});
