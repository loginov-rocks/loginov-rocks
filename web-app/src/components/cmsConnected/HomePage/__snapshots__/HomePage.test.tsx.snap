// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <h1>
    Hello!
  </h1>
  <nav>
    Home
     
    /
     
    <Link
      to="/education"
    >
      Education
    </Link>
  </nav>
</Fragment>
`;
